executionlayer/src/main/kotlin/maru/executionlayer/client/ShanghaiWeb3JJsonRpcExecutionLayerEngineApiClient.kt

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Consensys Software Inc.
+ *
+ * This file is dual-licensed under either the MIT license or Apache License 2.0.
+ * See the LICENSE-MIT and LICENSE-APACHE files in the repository root for details.
+ *
+ * SPDX-License-Identifier: MIT OR Apache-2.0
+ */
+package maru.executionlayer.client
+
+import java.util.Optional
+import maru.config.consensus.ElFork
+import maru.core.ExecutionPayload
+import maru.extensions.captureTimeSafeFuture
+import maru.mappers.Mappers.toDomainExecutionPayload
+import maru.mappers.Mappers.toExecutionPayloadV2
+import net.consensys.linea.metrics.MetricsFacade
+import tech.pegasys.teku.ethereum.executionclient.schema.ForkChoiceStateV1
+import tech.pegasys.teku.ethereum.executionclient.schema.ForkChoiceUpdatedResult
+import tech.pegasys.teku.ethereum.executionclient.schema.PayloadAttributesV1
+import tech.pegasys.teku.ethereum.executionclient.schema.PayloadAttributesV2
+import tech.pegasys.teku.ethereum.executionclient.schema.PayloadStatusV1
+import tech.pegasys.teku.ethereum.executionclient.schema.Response
+import tech.pegasys.teku.ethereum.executionclient.web3j.Web3JClient
+import tech.pegasys.teku.infrastructure.async.SafeFuture
+import tech.pegasys.teku.infrastructure.bytes.Bytes8
+
+// https://github.com/ethereum/execution-apis/blob/main/src/engine/shanghai.md
+class ShanghaiWeb3JJsonRpcExecutionLayerEngineApiClient(
```

Comment:
It would be nice to test it with a real node. For the `PragueWeb3JJsonRpcExecutionLayerEngineApiClient` it was done by `EagerQbftBlockCreatorTest`, because `EagerQbftBlockCreator` is supposed to implement the correct sequence of calls, which serves as a test for the client itself.

It's Ok to do it in a follow-up PR